### ISSUE TITLE
Feature Request: Allow customize checkout selection column definition

### ISSUE THREAD

**Issue by xiaoyu-tamu** [mui-org/material-ui-x#3420]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Summary 💡

https://github.com/mui-org/material-ui-x/blob/master/packages/grid/_modules_/grid/models/colDef/gridCheckboxSelectionColDef.tsx#L13

Is there anyway to customize checkout selection cell width? 

### Examples 🌈

![image](https://user-images.githubusercontent.com/33362998/145729146-fa3fd4e4-577b-472a-8dc0-f01b691b5148.png)

### Motivation 🔦

I want to change the minWidth and maxWidth to different value to keep the cell looks like a square when density is not standard.

**Comment by flaviendelangle** [comment]
Not currently,

For the grouping features, I have introduced a `groupingColDef` prop to allow overrides of the grouping column(s)
I intend to do the same for the selection.

In the meantime, the width is a property used a lot on the Grid so I would advise against overriding it with CSS, you would risk inconsistent behaviors.

**Comment by m4theushw** [comment]
A bit related to https://github.com/mui-org/material-ui-x/issues/2380#issuecomment-900420834. Users might want to have the checkboxes column at the end too. Instead of adding a prop, like `groupingColDef`, we could allow users to reuse `GRID_CHECKBOX_SELECTION_COL_DEF` and pass the customized column in the `columns` prop. I guess the only thing needed to allow this is to not add a second column if there's already one, during hydration.

**Comment by flaviendelangle** [comment]
If I understand correctly, you would do something like "if a column has the field of `GRID_CHECKBOX_SELECTION_COL_DEF` then don't re-add it" ?

**Comment by xiaoyu-tamu** [comment]
> it also add the ability to disable hidden for checkout selection.


![image](https://user-images.githubusercontent.com/33362998/145864531-451f2226-4893-4b6a-aa48-f6f9c64dcd75.png)

what i mean is to disabled column selector for selection cell, but i just realized there is not such functionality :-/ 

i should open another feature request for Fine-tune disabled column selector column level

**Comment by m4theushw** [comment]
> If I understand correctly, you would do something like "if a column has the field of GRID_CHECKBOX_SELECTION_COL_DEF then don't re-add it" ?

@flaviendelangle Yes, using the `columns` prop we solve two problems at once: customizability and position change.

---

> My use case is to not allow user hide "Checkbox Selection Cell"

I would extend this use case to allow blocking any column from being hidden. I could have an "actions" column that I need to keep always visible. I would propose to add a `hideable` property to `GridColDef` to allow to configure it per column. AG Grid has `lockVisible`.

![image](https://user-images.githubusercontent.com/42154031/145904934-014a2783-1355-4853-b3e1-34b4869f3dc9.png)

@xiaoyu-tamu If you could create another issue only for this last request it would be awesome. There we can discuss the right naming.

**Comment by xiaoyu-tamu** [comment]
#3426 created another feature request for hideable prop

**Comment by flaviendelangle** [comment]
The customization / position of the checkbox column already works
See [this example](https://codesandbox.io/s/data-grid-community-forked-ifb49w?file=/src/App.tsx)

We only have a problem with tree data and row grouping which moves the grouping column(s) at the 2nd position starting left when there is a checkbox column.
So if we move the checkbox column, the grouping column(s) will be mis-placed.
I can change the check to say "if the 1st column is the checkbox column, the place the grouping column right of it, else place it at the very left"

**Comment by hoangquoclk** [comment]
![column](https://github.com/mui/mui-x/assets/54970942/c0a5e057-2639-4800-af14-5b589e7980ee)
I use this object to define a checkbox (with field: '__check__'). We can set minWidth and maxWidth for it
